src/request-response-types.ts

```diff
@@ -2,28 +2,35 @@
 
 import {
    APIGatewayEventRequestContext as OrigAPIGatewayEventRequestContext,
+   APIGatewayEventRequestContextV2 as OrigAPIGatewayEventRequestContextV2,
    APIGatewayProxyEvent,
+   APIGatewayProxyEventV2,
    Context,
    APIGatewayProxyResult,
+   APIGatewayProxyStructuredResultV2,
+   ALBEvent,
+   ALBEventRequestContext,
+   ALBResult,
 } from 'aws-lambda';
-import { StringMap, StringArrayOfStringsMap } from '@silvermine/toolbox';
 
 /* COMBO TYPES */
 
 /**
  * The `evt` argument passed to a Lambda handler that represents the request (from API
  * Gateway or ALB).
  */
-export type RequestEvent = ApplicationLoadBalancerRequestEvent | APIGatewayRequestEvent;
+export type RequestEvent = ApplicationLoadBalancerRequestEvent | APIGatewayRequestEvent | APIGatewayRequestEventV2;
 
 /**
  * The "request context", which is accessible at `evt.requestContext`.
  */
 export type RequestEventRequestContext = APIGatewayEventRequestContext | ApplicationLoadBalancerEventRequestContext;
 
-export interface ResponseResult extends APIGatewayProxyResult {
-   multiValueHeaders: StringArrayOfStringsMap;
-   statusDescription?: string;
+export type ResponseResult = APIGatewayProxyResult | APIGatewayProxyStructuredResultV2 | ALBResult;
+
+export function isALBResult(evt: ResponseResult, test: boolean): evt is ALBResult {
+   // TODO - this type gaurd doesn't do any useful checking
+   return test && 'statusCode' in evt;
 }
```

Comment:
Flagging this TODO for feedback on this approach to using a type guard. Doesn't seem to be any attributes that can consistently be used to differentiate between the types. So seems like we can either:
A. Use a type guard like this.
B. Extend ALBResult to have an attribute we set to indicate if it is ALB, and then base the type guard on that attribute.